### ISSUE TITLE
gfx_window_glutin::update_views displays incorrectly when resizing window vertically

### ISSUE THREAD

**Issue by Mistodon** [gfx-rs/gfx#2068]
Short info header:
- GFX version: 0.17.1
- OS: macOS 10.13.3
- GPU: Intel Iris Plus Graphics 640 1536 MB (Macbook Pro)

Using gfx_window_glutin (v0.23.0) and calling `update_views` after resizing the window doesn't seem to work properly when resizing it vertically.

This image shows the initial window, the same window resized horizontally, and then vertically. The coordinate system stretches correctly for the second image, but not for the third.

<img width="289" alt="resize_bug" src="https://user-images.githubusercontent.com/1149443/40545486-e6ab611c-6023-11e8-9c98-9f5a5af42045.png">

Using `init_existing` instead and replacing the device, factory, color_view, and depth_view values does not exhibit this problem.

**Comment by kvark** [comment]
@Mistodon is the issue reproducible for you on the `cube` example?
`update_views` works within assumption that you will not use the old views anywhere. We should probably add more run-time checks to assert on that. But if you still have the old views in some pipeline data structs, you'll get incorrect window updates.

**Comment by kvark** [comment]
@francesca64 oh, good to know! Please come visit us on gitter in case you have any issues.
I agree that the framebuffer RTVs on GL are not handled in the cleanest way... not enough engineering effort has been put into this, and everyone went ~~to the front~~ to build HAL instead.

**Comment by Mistodon** [comment]
I've tested the triangle and cube examples and they work fine, so I must be doing something wrong. I'm still not sure quite what, but I'll take it to gitter :).

**Comment by kvark** [comment]
Yeah, sounds like some RTV handles are stale on your side, closing this.

**Comment by Mistodon** [comment]
Almost as soon as this was closed, I finally found what was causing the issue - even if I don't 100% understand why.

For anyone that might run into this in future: I was missing a call to [`window.resize(w, h)`](https://docs.rs/glutin/0.15.0/glutin/trait.GlContext.html#tymethod.resize) in my update event. I didn't realize this had to be done manually, but removing it from the triangle example reproduces the exact issue I was seeing, and adding it fixed my problem.
